### PR TITLE
Order dev

### PR DESCRIPTION
**Pull request by JuanGQCadavid**

### CODE REVIEW

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        List payload = ex.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map(x -> {
+                    return new Pair(x.getField(),x.getDefaultMessage());
+                })
+                .collect(Collectors.toList());
+        String errorMessage = "There is a problem with the fields format.";
+
+        return new ResponseEntity<>(new ErrorMessage(errorMessage, new GeneralErrorType("MissingFieldsBody",HttpStatus.BAD_REQUEST),payload),HttpStatus.BAD_REQUEST);
```
Comment (czuluagah) [inline utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java]: Create a Constant class to manage all the hardcoded strings

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/ErrorMessage.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Error class that represents the error itself
+ * it contains the timeStamp, the message and a
+ * payLoad that shows where the error is.
+ *
+ * Also, it has a variable that represents the internal
+ * error for the user.
+ */
+
+@Getter
```
Comment (czuluagah) [inline utilities/src/main/java/com/ecommercepractice/utilities/exception/ErrorMessage.java:17]: Use Builder patterm

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * General error handler for all the Order exceptions family.
+     * @param ex
+     * @return
+     */
+    @ExceptionHandler(GeneralException.class)
+    public ResponseEntity<ErrorMessage> orderExceptionsHandler(GeneralException ex){
+        log.error(ex.getMessage());
+        return new ResponseEntity( new ErrorMessage(ex), ex.getGeneralErrorType().getStatus());
```
Comment (czuluagah) [inline utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java]: Use Builder Patterm

File: utilities/src/main/java/com/ecommercepractice/utilities/App.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.ecommercepractice.utilities;
+
```
Comment (czuluagah) [inline utilities/src/main/java/com/ecommercepractice/utilities/App.java]: Remove this class

File: utilities/0.0.1-SNAPSHOT/utilities-0.0.1-SNAPSHOT.pom [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.2.7.RELEASE</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.e-commerce-practice</groupId>
+	<artifactId>utilities</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (czuluagah) [inline utilities/0.0.1-SNAPSHOT/utilities-0.0.1-SNAPSHOT.pom]: use your version convention

Comment (JuanGQCadavid) [reply utilities/0.0.1-SNAPSHOT/utilities-0.0.1-SNAPSHOT.pom]: OK

File: utilities/src/main/java/com/ecommercepractice/utilities/App.java (hunk 4 above)
Comment (JuanGQCadavid) [reply utilities/src/main/java/com/ecommercepractice/utilities/App.java]: Ok

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java (hunk 1 above)
Comment (JuanGQCadavid) [reply utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java]: OK

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/ErrorMessage.java (hunk 2 above)
Comment (JuanGQCadavid) [reply utilities/src/main/java/com/ecommercepractice/utilities/exception/ErrorMessage.java:17]: Done

File: utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java (hunk 3 above)
Comment (JuanGQCadavid) [reply utilities/src/main/java/com/ecommercepractice/utilities/exception/GeneralExceptionHandler.java]: Done

**Pull request merged**